internal/ethapi/api.go

```diff
@@ -1642,10 +1642,7 @@ func (s *TransactionAPI) GetTransactionByHash(ctx context.Context, hash common.H
 		if tx := s.b.GetPoolTransaction(hash); tx != nil {
 			return NewRPCPendingTransaction(tx, s.b.CurrentHeader(), s.b.ChainConfig()), nil
 		}
-		if err == nil {
-			return nil, nil
-		}
-		return nil, NewTxIndexingError()
+		return nil, err
```

Comment:
It's wrong. Originally if the transaction is not found and background indexing is not running, null is returned.

Now, we return whatever error we have to the RPC consumers. It's a behavioral change.